### PR TITLE
Having doughnut support in extrinsic

### PR DESCRIPTION
**Pull request by KarishmaBothara**
doughnut_in_extrinsic

Use the doughnut bytes from doughnut-maker or use branch https://github.com/cennznet/api.js/pull/126 and use tests to create one
sample - 0x000000d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d8eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a48dfa46d5563656e6e7a6e6574000000000000000046000000000067656e657269632d617373657400000000000000000000000000000000000000007472616e736665720000000000000000000000000000000000000000000000000096099ffb7e20b8c41676e6774d6f116ba1819a9281f2b85e5daa9ca3526bc864d1a55f16f4d07d80c5daca46ea7dfb226dbc95c9e4b1a5f5ac55a0af8f6d3389

### CODE REVIEW

File: packages/react-signer/src/Modal.tsx [hunk 1]
```diff
@@ -656,6 +658,10 @@ class Signer extends React.PureComponent<Props, State> {
       params.push({ tip } as Partial<SignerOptions>);
     }
 
+    if (doughnut) {
+      params.push({doughnut:new Doughnut(api.registry, hexToU8a(doughnut))})
```
Comment (holygits) [inline packages/react-signer/src/Modal.tsx:662]: formatting
```suggestion
      params.push({ doughnut: new Doughnut(api.registry, hexToU8a(doughnut)) })
```

File: packages/apps/webpack.config.js [hunk 2]
```diff
@@ -156,6 +156,7 @@ function createWebpack ({ alias = {}, context, name = 'index' }) {
       new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
       new webpack.DefinePlugin({
         'process.env': {
+          DOUGHNUT_SUPPORT: JSON.stringify(true),
```
Comment (holygits) [inline packages/apps/webpack.config.js:159]: should set it to `false` by default or alternatively disable it on production build i.e use `isProd`

File: packages/react-signer/src/Modal.tsx (hunk 1 above)
Comment (KarishmaBothara) [reply packages/react-signer/src/Modal.tsx]: done

File: packages/apps/webpack.config.js (hunk 2 above)
Comment (KarishmaBothara) [reply packages/apps/webpack.config.js]: done.

**Comment by benxgao** [comment]
@KarishmaBothara I am testing the branch in local, and not sure if I still need to follow instructions as in PR description, or any updates? Is the release of api@1.1.0 supporting this branch?

**Comment by KarishmaBothara** [comment]
> @KarishmaBothara I am testing the branch in local, and not sure if I still need to follow instructions addressed in PR description, or is there any update?

Updated the description.. this branch uses new api release and the doughnut-wasm package..

File: packages/react-signer/src/Modal.tsx [hunk 3]
```diff
@@ -656,6 +658,10 @@ class Signer extends React.PureComponent<Props, State> {
       params.push({ tip } as Partial<SignerOptions>);
     }
 
+    if (doughnut) {
+      params.push({ doughnut: new Doughnut(api.registry, hexToU8a(doughnut)) })
```
Comment (benxgao) [inline packages/react-signer/src/Modal.tsx:662]: Missing semicolon.

**Pull request merged**